src/db.c

```diff
@@ -1353,6 +1355,11 @@ void shutdownCommand(client *c) {
         return;
     }
 
+    if (flags & SHUTDOWN_FAILOVER && !server.cluster_enabled) {
```

Comment:
i see we check this, so we may also need to check cluster in config.c

Reply:
We didn't check it before... I just wanted to change the config, not implement more stuff. :) But I guess you are right. It's better to check it.

Reply:
If we forbid it in config if cluster mode is disabled, it means Valkey refuses to start if this config is set. Maybe it's not good if some users want to just change cluster-enabled to enable/disable cluster mode. They would need to change multiple configs at the same time. This kind of dependency between configs is annoying, WDYT?

Reply:
sure, that is not a big deal, i am ok with this.

Reply:
Do we have a plan (I thought we do?) to extend this feature to standalone? if yes, the current implementation makes sense.

Reply:
on a side note, it is a bit odd to see `shutdownCommand` implemented in `db.c`.

Reply:
Yes, I've opened an issue for standalone:

* #2349

db.c contains a lot of various stuff. We should maybe organize it better but not in this PR i hope.